### ISSUE TITLE
Moderated comments shouldn't be *NEW*

### ISSUE THREAD

**Issue by hljonker** [SoylentNews/rehash#404]
I just moderated a post and noticed that afterwards, the post was still marked as "NEW".
That should be an error - I shouldn't be moderating posts that I have not read.

Reporting it here only because I feel that the Power That Be care about making SN better in every sense, even if it is as tiny an improvement as this.
If I could set priority, it'd be "basement".

**Comment by TheMightyBuzzard** [comment]
Yup. That happens intentionally because of the noupdate=1 in the url. We don't want to mark everything older than the comment you just moderated as read because threading means comments below where you are on the page could be older or newer than the one you just moderated. There's no way to tell without putting in a lot of extra processing.

We could technically check on a comment-by-comment basis but at best that would cause a significant performance hit by requiring an extra db call for each comment. We talked about doing that back when we were redoing the comments but ended up deciding it was better to be responsive than 100% accurate.

That said, I'll give it another look anyway when I get time and see if I can figure out a way to do it with just one db call and some minimal processing.
